src/platform/linux/x11/mod.rs

```diff
@@ -142,25 +135,40 @@ impl EventsLoop {
                 (xlib.XNextEvent)(self.display.display, &mut xev);
             }
             self.process_event(&mut xev, &mut callback);
-            if self.interrupted.load(::std::sync::atomic::Ordering::Relaxed) {
-                break;
-            }
         }
     }
 
-    pub fn run_forever<F>(&self, mut callback: F)
-        where F: FnMut(Event)
+    pub fn run_forever<F>(&mut self, mut callback: F)
+        where F: FnMut(Event) -> ControlFlow
     {
-        self.interrupted.store(false, ::std::sync::atomic::Ordering::Relaxed);
+        self.pending_wakeup.store(false, atomic::Ordering::Relaxed);
 
         let xlib = &self.display.xlib;
 
         let mut xev = unsafe { mem::uninitialized() };
 
         loop {
             unsafe { (xlib.XNextEvent)(self.display.display, &mut xev) }; // Blocks as necessary
-            self.process_event(&mut xev, &mut callback);
-            if self.interrupted.load(::std::sync::atomic::Ordering::Relaxed) {
+
+            let mut control_flow = ControlFlow::Continue;
+
+            if self.pending_wakeup.load(atomic::Ordering::Relaxed) {
```

Comment:
Oh my, I'm so sorry. I didn't look at this as thoroughly as I should have when making this comment. I didn't see originally that the actual event loop bump was moved rather than deleted.